demo/scripts/controlsV2/demoButtons/imageCropButton.ts

```diff
@@ -0,0 +1,26 @@
+import type { RibbonButton } from '../roosterjsReact/ribbon';
+
+/**
+ * @internal
+ * "Crop Image" button on the format ribbon
+ */
+export const imageCropButton: RibbonButton<'buttonNameCropImage'> = {
+    key: 'buttonNameCropImage',
+    unlocalizedText: 'Crop Image',
+    iconName: 'Crop',
+    isDisabled: formatState => !formatState.canAddImageAltText,
+    onClick: editor => {
+        const selection = editor.getDOMSelection();
+        if (selection?.type === 'image') {
+            editor.triggerEvent('editImage', {
```

Comment:
Please use a more former way to trigger this action. You can follow the existing pattern in image edit context menu which takes `ImageEditor` interface as parameter so that in all action code it can directly call function from this interface. Then you need to make your plugin implement this interface.

Reply:
And actually this changed the meaning of existing editImage event. Its original meaning is the image is already modified, so plugins can do their own work accordingly, but here it means let plugin edit the image and the image is not changed at this point.